freedomofpress/securedrop-workstation
# Fixes dom0 updates in GUI updater

## Status

Ready for review 

## Description of Changes

Fixes #646 

Changes proposed in this pull request:

Actually runs the package updates within dom0, by iterating over the
generator returned by the apply_updates function, ensuring the logic
runs. Revises the progress bar logic a bit, since the apply_updates
function is called twice, with different expectations about how much the
progress bar should increase.


## Testing
Ideally, install 0.5.0 prod (that's n-1 from most current stable release). Then:

1. Clone this branch to dom0
2. `sudo cp -v launcher/sdw_updater_gui/Updater{,App}.py /opt/securedrop/launcher/sdw_updater_gui/`
3. Close the updater GUI window if it is already open, to ensure a fresh start
4. Confirm via `dnf info securedrop-workstation-dom0-config is at 0.5.0, not 0.5.1
5. Confirm via `qvm-ls | grep fedora-32` that fedora-32 is not installed on the system
6. Run `time /opt/securedrop/launcher/sdw-launcher.py --skip-delta 0` and proceed with upgrade
7. Ensure that 0.5.1 RPM is installed
8. Ensure that F32 is installed and set for all `sys-*` VMs
9. Ensure no errors


## Checklist

### If you have made code changes

      be left unchecked, as `flake8` also runs in CI)

### If you have made changes to the provisioning logic


      logic in `MANIFEST.in` and `rpm-build/SPECS/securedrop-workstation-dom0-config.spec`
